lib/lf/fe/loc_comp_ellbvp.h

```diff
@@ -241,367 +263,454 @@ class LagrangeFEEllBVPElementMatrix {
-       }
-       return mat;
-       break;
-     }
-     case lf::base::RefEl::kQuad(): {
-       // Quadrature points in actual cell
-       const Eigen::MatrixXd mapped_qpts(geo_ptr->Global(qr_quad_.Points()));
-       // Obtain the metric factors for the quadrature points
-       const Eigen::VectorXd determinants(
-           geo_ptr->IntegrationElement(qr_quad_.Points()));
-       // Fetch the transformation matrices for the gradients
-       const Eigen::MatrixXd JinvT(
-           geo_ptr->JacobianInverseGramian(qr_quad_.Points()));
-
-       // Element matrix
-       ElemMat mat(Nrsf_quad_, Nrsf_quad_);
-       mat.setZero();
-
-       // Loop over quadrature points
-       for (int k = 0; k < Nqp_quad_; ++k) {
-         // Evaluate diffusion and reaction coefficient
-         // at quadrature points in actual cell
-         const auto alphaval{alpha_(mapped_qpts.col(k))};
-         const auto gammaval{gamma_(mapped_qpts.col(k))};
-
-         const double w = qr_quad_.Weights()[k] * determinants[k];
-         // Transformed gradients
-         const auto trf_grad(JinvT.block(0, 2 * k, 2, 2) *
-                             grad_quadpoint_quad_[k]);
-         // Transformed gradients multiplied with coefficient
-         const auto alpha_trf_grad(alphaval * trf_grad);
-         mat += w * (alpha_trf_grad.transpose() * trf_grad +
-                     (gammaval * rsf_quadpoints_quad_.col(k)) *
-                         (rsf_quadpoints_quad_.col(k).transpose()));
-       }
-       return mat;
-       break;
-     }
-     default: { LF_ASSERT_MSG(false, "Illegal cell type"); }
-   }  // end switch
-   return Eigen::MatrixXd(0, 0);
- }
+
+ private:
+  /** @brief Predicate telling whether an edge should be taken into account
+     during the assembly of the Galerkin matrix */
+  EDGESELECTOR edge_sel_;
+};
 
 /**
- * @brief Local computation of general element vector for scalar finite elements
+ * @brief Local computation of general element (load) vector for scalar finite
+ * elements; volume contributions only
  *
  * @tparam SCALAR underlying scalar type, usually double or complex<double>
  * @tparam FUNCTOR object with an evaluation operator of signature
- *         std::function<double(const Eigen::Vector2d &)>, which supplies
+ *         std::function<SCALAR(const Eigen::VectorXd &)>, which supplies
  *         the source function
  *
+ * The underlying local linear form is
+ * @f[
+      v \mapsto \int_K f(\mathbf{x})\,v(\mathbf{x}\,\mathrm{d}\mathbf{x}\;,
+ * @f]
+ * where \f$f\f$ is suppoed to be a locally continuous source function.
+ *
  * Computation is based on a quadrature rules supplied by the LehrFEM++
  * lf::quad::QuadRule module.
  *
  * This class complies with the requirements for the template parameter
  * `ELEM_VEC_COMP` of the function AssembleVectorLocally().
  */
 template <typename SCALAR, typename FUNCTOR>
-class ScalarFELocalLoadVector {
+class ScalarFELocalLoadVector : public LocCompLagrFEPreprocessor {
```

Comment:
Same thing here: why is a `ScalarFELocalLoadVector` a `LocCompLagrFEPreprocessor`?